spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/ReactivePulsarReaderFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core.reactive;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageReader;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageReaderSpec;
+
+/**
+ * The strategy to create a {@link ReactiveMessageReader} instance(s).
+ *
+ * @param <T> reactive message reader payload type
+ * @author Christophe Bornet
+ */
+public interface ReactivePulsarReaderFactory<T> {
+
+	/**
+	 * Create a reactive message reader.
+	 * @param schema the schema of the messages to be read
+	 * @param spec a reader spec to override the defaults of the factory
+	 * @return the reactive message reader
+	 */
+	ReactiveMessageReader<T> createReader(Schema<T> schema, ReactiveMessageReaderSpec spec);
```

Comment:
Whatever we decide on the consumer factory wrt to convenience method w/o `spec` will apply here as well.